daemons/gptp/windows/daemon_cl/tsc.hpp

```diff
@@ -50,30 +49,52 @@ inline unsigned __int64 PLAT_rdtsc()
 	return __rdtsc();
 }
 
+#define CLOCK_INFO_CPUID_LEAF	(0x15)
+#define SYSTEM_CLOCK_24MHZ		(24000000)
+
 /**
  * @brief  Gets the TSC frequnecy
  * @param  millis time in miliseconds
- * @return TSC frequency
+ * @return TSC frequency, or 0 on error
  */
 inline uint64_t getTSCFrequency( unsigned millis ) {
-	UINT16 multiplierx2;
-	uint64_t frequency = 0;
-	DWORD mhz;
-	DWORD mhz_sz = sizeof(mhz);
+	int max_cpuid_level;
+	int tmp[4];
+
+	// Find the max cpuid level, and if possible find clock info
+	__cpuid(tmp, 0);
+	max_cpuid_level = tmp[0];
+	if (max_cpuid_level >= CLOCK_INFO_CPUID_LEAF)
+		__cpuid(tmp, CLOCK_INFO_CPUID_LEAF);
 
-	if( RegGetValue( HKEY_LOCAL_MACHINE, "HARDWARE\\DESCRIPTION\\System\\CentralProcessor\\0", "~MHz", RRF_RT_REG_DWORD, NULL, &mhz, &mhz_sz ) != ERROR_SUCCESS ) {
-		goto done;
+	// For pre-Win10 on 6th Generation or greater Intel CPUs the raw hardware
+	// clock will be returned, *else* use QPC for everything else
+	//
+	// EAX (tmp[0]) must be >= 1, See Intel SDM 17.15.4 "Invariant Time-keeping"
+	if (!IsWindows10OrGreater() && max_cpuid_level >= CLOCK_INFO_CPUID_LEAF &&
+		tmp[0] >= 1) {
+		SYSTEM_INFO info;
+
+		GetSystemInfo(&info);
+
+		// Look at processor model number to determine clock rate
+		switch (info.wProcessorRevision >> 8)
+		{
+		default:
+			// Unable to identify CPU
+			return 0;
+		case 0x5E:
```

Comment:
I understand by the code that these values (0x5e and 0x4e) are processor model identifiers, but I think it would improve readability if we had a define (or a const static uint8_t variable somewhere) for them.